tests/scalers/cpu/cpu_test.go

```diff
@@ -204,6 +206,15 @@ func scaleOut(t *testing.T, kc *kubernetes.Clientset, data templateData) {
 	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 1, 60, 1),
 		"Replica count should start out as 1")
 
+	// The default metrics-server window is 30s, and that's what keda is used to, but on OpenShift we use
+	// prometheus-adapter, and we have it tuned to a window of 5m, so on OpenShift it takes 5 minutes before
+	// the HPA can even start scaling, and we fail the test unless we wait for the metrics. We'd read the
+	// window straight from the adapter-config configmap in openshift-monitoring, but then the test would be
+	// tied to a specific metrics server implementation, so we just wait up to 10 minutes for the metrics before
+	// we proceed with the test.
+	require.True(t, WaitForHPAMetricsToPopulate(t, kc, deploymentName, testNamespace, 120, 5),
```

Comment:
This would be great addition to upstream imho. (Just remove the openshift specific comments).

Reply:
Thanks @zroubalik ! I made it generic and put up: https://github.com/kedacore/keda/pull/5294, I can adjust it however we need to.